numpy/core/src/multiarray/arrayobject.c

```diff
@@ -453,6 +442,31 @@ PyArray_ResolveWritebackIfCopy(PyArrayObject * self)
 
 /*********************** end C-API functions **********************/
 
+
+/* dealloc must not raise an error, best effort try to write
+   to stderr and clear the error
+*/
+
+static NPY_INLINE void
+WARN_IN_DEALLOC(PyObject* warning, const char * msg) {
+    if (PyErr_WarnEx(warning, msg, 1) < 0) {
+        PyObject * s;
+
+#if PY_MAJOR_VERSION < 3
+        s = PyString_FromString(msg);
+#else
+        s = PyUnicode_FromString(msg);
+#endif
+        if (s) {
+            PyErr_WriteUnraisable(s);
```

Comment:
I think I mentioned this last time too, but `s` should not be the error message, as that is already fetched from the raised warning:

> The function is called with a single argument obj that identifies the context in which the unraisable exception occurred. If possible, the repr of obj will be printed in the warning message.

As you have it right now, the message appears twice

Reply:
fixed, now prints ``Exception RuntimeWarning: RuntimeWarning('ERROR: WRITEBACKIFCOPY detected in array_dealloc.  Required call to PyArray_ResolveWritebackIfCopy or PyArray_DiscardWritebackIfCopy is missing. This could also be caused by using a nditer without a context manager',) in 'array_dealloc' ignored``, where the last ``'array_dealloc'`` comes from the ``s`` string